airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -16,299 +16,793 @@
+            will check the status of any SageMaker job
+        :type check_interval: int
+        :param max_ingestion_time: the maximum ingestion time in seconds. Any
+            SageMaker jobs that run longer than this will fail. Setting this to
+            None implies no timeout for any SageMaker job.
+        :type max_ingestion_time: int
+        :return: A response to endpoint update
         """
 
-        return self.conn.create_model(
-            **model_config)
+        response = self.get_conn().update_endpoint(**config)
+        if wait_for_completion:
+            self.check_status(config['EndpointName'],
+                              SageMakerHook.non_terminal_states,
+                              SageMakerHook.failed_states,
+                              'EndpointStatus',
+                              self.describe_endpoint,
+                              check_interval, max_ingestion_time
+                              )
+        return response
 
-    def describe_training_job(self, training_job_name):
+    def describe_training_job(self, name):
         """
-        :param training_job_name: the name of the training job
-        :type training_job_name: str
-        Return the training job info associated with the current job_name
+        Return the training job info associated with the name
+
+        :param name: the name of the training job
+        :type name: str
         :return: A dict contains all the training job info
         """
-        return self.conn\
-                   .describe_training_job(TrainingJobName=training_job_name)
 
-    def describe_tuning_job(self, tuning_job_name):
+        return self.get_conn().describe_training_job(TrainingJobName=name)
+
+    def describe_training_job_with_log(self, job_name, non_terminal_states, positions, stream_names,
+                                       instance_count, state, last_description,
+                                       last_describe_job_call):
+        """
+        Return the training job info associated with job_name and print CloudWatch logs
+        """
+        log_group = '/aws/sagemaker/TrainingJobs'
+
+        if len(stream_names) < instance_count:
+            # Log streams are created whenever a container starts writing to stdout/err, so this list
+            # may be dynamic until we have a stream for every instance.
+            try:
+                streams = self.get_log_conn().describe_log_streams(
+                    logGroupName=log_group,
+                    logStreamNamePrefix=job_name + '/',
+                    orderBy='LogStreamName',
+                    limit=instance_count
+                )
+                stream_names = [s['logStreamName'] for s in streams['logStreams']]
+                positions.update([(s, Position(timestamp=0, skip=0))
+                                  for s in stream_names if s not in positions])
+            except ClientError as e:
```

Comment:
I think we can do this as:

```
logs_conn = self.get_log_conn()
try:
    ....
except logs_conn.exceptions.ResourceNotFoundException:
    pass
```

(I think ew're on a recent enough version of boto3/botocore for this to work)

Reply:
Updated.